src/AutoRest.CSharp.V3/Generation/Writers/CodeWriter.cs

```diff
@@ -190,6 +190,7 @@ private void AppendType(CSharpType type)
             {
                 UseNamespace(type.Namespace);
 
+                AppendRaw("global::");
```

Comment:
Nice. We ended up doing the same thing in the low-level C# generator for Azure PowerShell.